### DISCUSSION TITLE
Integrate nanoDEP with nanoMDM

### DISCUSSION THREAD

**Discussion by ashishmeher216** [micromdm/nanodep#61] [General]
I am using nanoMDM and it is working well. Great product.  I have an ABM account. Is it possible to integrate nanoDEP with nanoMDM smoothly? Or do I have to manually do that?

**Comment by jessepeterson** [comment]
> Is it possible to integrate nanoDEP with nanoMDM smoothly? Or do I have to manually do that?

Hello @ashishmeher216. What do you mean by "integrate?" Is there something specific you'd like to see work better between the two? ABM/DEP access is somewhat distinct from an MDM server and both NanoDEP and NanoMDM have "minimalist" approaches (leaving tight/opinionated coupling up to you, generally speaking). I'm curious to know what you're thinking! Thanks! :)

**Reply by ashishmeher216** [reply]
Hey @jessepeterson, I want to know the difference between User Enrollment flow present in NanoMDM and the way NanoDEP handles it. If someone has a BYOD, I would enroll his iOS device using the User Enrollment channel by using managed Apple IS from ABM right?  NanoDEP with ABM  would only be for my company-owned devices.  I wonder if I am correct xd.

One more thing, from NanoMDM, what do all these enrollment types mean?
![image](https://github.com/micromdm/nanodep/assets/51280607/499aef2e-46c6-469e-9d90-eae864576a29)

Would love to hear from you :)

**Reply by jessepeterson** [reply]
I think you have that right, yes. BYOD (User Enrollment) requires an Apple ID enrollment flow that you have to build.

The listed enrollment types that you have in the picture are different MDM "channels" — different enrollment types that can be targeted specifically (even if on some devices, like macOS and iPads in Shared iPad mode they can be on the same device).

**Reply by ashishmeher216** [reply]
Is the BYOD (User Enrollment) flow, which necessitates an Apple ID, absent from both NanoMDM and NanoDEP?

**Reply by ashishmeher216** [reply]
Hey @jessepeterson could you please provide some insight?

**Reply by jessepeterson** [reply]
"BYOD" (User Enrollment) enrollment types have worked great with NanoMDM. Previously you were able to just add a Managed Apple ID to the enrollment profile and enroll that way. However Apple is transitioning to the User Enrollment "flow" which includes the `.well-known` address [which they've documented here]( https://developer.apple.com/documentation/devicemanagement/user_enrollment/onboarding_users_with_account_sign-in/implementing_the_simple_authentication_user-enrollment_flow). This is similar to the new Account-Driven Device Enrollment (ADDE) [flow](https://developer.apple.com/wwdc23/10254), too.

As far as the Micro/Nano projects: How you serve your enrollment process is not in scope for the project(s). That's something you build yourself. Including any User Enrollment flow. Not to mention for an Managed Apple ID on a specific domain name you'd need to have the `.well-known` portion hosted *on **your** (or your customer's) website* anyway. Hope that's helpful!
